ts/lib/shortcuts.ts

```diff
@@ -0,0 +1,149 @@
+// Copyright: Ankitects Pty Ltd and contributors
+// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+import * as tr from "./i18n";
+
+export type Modifier = "Control" | "Alt" | "Shift" | "Meta";
+
+const modifiers: Modifier[] = ["Control", "Alt", "Shift", "Meta"];
+
+// how modifiers are mapped
+const platformModifiers =
+    navigator.platform === "MacIntel"
+        ? ["Meta", "Alt", "Shift", "Control"]
+        : ["Control", "Alt", "Shift", "OS"];
+
+function splitKeyCombinationString(keyCombinationString: string): string[][] {
+    return keyCombinationString.split(", ").map((segment) => segment.split("+"));
+}
+
+function modifiersToPlatformString(modifiers: string[]): string {
+    const displayModifiers =
+        navigator.platform === "MacIntel"
+            ? ["^", "⌥", "⇧", "⌘"]
+            : [`${tr.keyboardCtrl()}+`, "Alt+", `${tr.keyboardShift()}+`, "Win+"];
+
+    let result = "";
+
+    for (const modifier of modifiers) {
+        result += displayModifiers[platformModifiers.indexOf(modifier)];
+    }
+
+    return result;
+}
+
+const alphabeticPrefix = "Key";
+const numericPrefix = "Digit";
+const keyToCharacterMap = {
+    Backslash: "\\",
+    Backquote: "`",
+    BracketLeft: "[",
+    BrackerRight: "]",
+    Quote: "'",
+    Semicolon: ";",
+    Minus: "-",
+    Equal: "=",
+    Comma: ",",
+    Period: ".",
+    Slash: "/",
+};
+
+function keyToPlatformString(key: string): string {
+    return key.startsWith(alphabeticPrefix)
+        ? key.slice(alphabeticPrefix.length)
+        : key.startsWith(numericPrefix)
+        ? key.slice(numericPrefix.length)
+        : Object.prototype.hasOwnProperty.call(keyToCharacterMap, key)
+        ? keyToCharacterMap[key]
+        : key;
```

Comment:
I'm surprised these are not being caught by https://eslint.org/docs/rules/no-nested-ternary - I think once there's more than one of them, the code would be more readable as an if/elseif, even if it's more verbose.

Reply:
My main reason why I prefer ternary expressions, is that you can be sure that you don't miss a return statement, or a pathway by accident. But I'm fine with turning into an if statement. Too bad we don't have syntax like in Rust here :(

Reply:
I do really miss implicit returns, and wasted half an hour the other day tracking down a bug that was caused by me not returning a value in a .map() call, that TypeScript didn't warn about because I had not specified the return type. :-( But in this case it would presumably be caught by the return type on the function, and be considerably more readable :-)